Move shell to use `truncate_text`

aibrahim-oai opened the pull request:
Move shell to use the configurable `truncate_text`

aibrahim-oai committed:
shell

aibrahim-oai committed:
tests

aibrahim-oai committed:
clean

aibrahim-oai commented:
@codex review this

aibrahim-oai committed:
clean

aibrahim-oai committed:
tests

aibrahim-oai committed:
tests

aibrahim-oai committed:
tests

aibrahim-oai committed:
tests

aibrahim-oai committed:
tests

aibrahim-oai committed:
tests

aibrahim-oai commented:
@codex review this

pakrym-oai commented on codex-rs/core/tests/suite/shell_serialization.rs:
```diff
@@ -449,26 +449,7 @@ async fn shell_output_reserializes_truncated_content(output_type: ShellModelOutp
-399
-400
-$"#;
+    let truncated_pattern = r#"(?s)(?:Total output lines: \d+\n\n)?\{"output":"1\\n2\\n3\\n4\\n5\\n6\\n.*?\[\u{2026}1910 tokens truncated\u{2026}].*?99996\\n99997\\n99998\\n99999\\n100000\\n","metadata":\{"exit_code":0,"duration_seconds":[0-9]+(?:\.[0-9]+)?\}"#;
```
This doesn't look right.

pakrym-oai commented on codex-rs/core/src/context_manager/history_tests.rs:
```diff
@@ -371,27 +371,19 @@ fn format_exec_output_truncates_large_error() {
     let large_error = line.repeat(2_500); // way beyond both byte and line limits
 
-    let truncated = truncate::truncate_with_line_bytes_budget(&large_error, EXEC_FORMAT_MAX_BYTES);
+    let truncated = truncate_exec_output(&large_error);
```
Why are we testing a private function that nobody uses?

pakrym-oai replied:
if these tests test truncate::truncate_text should they be in truncation.rs?

pakrym-oai commented on codex-rs/core/tests/suite/truncation.rs:
```diff
@@ -607,10 +589,99 @@ async fn large_budget_avoids_truncation() -> Result<()> {
+    let parsed: Value = serde_json::from_str(&output)?;
+    let expected_echo = format!("ECHOING: {large_msg}");
+    assert_eq!(parsed["echo"], expected_echo);
+    assert!(
```
assert the length of parsed output.

aibrahim-oai committed:
tests

aibrahim-oai committed:
shell-config

aibrahim-oai committed:
shell-config

aibrahim-oai committed:
Merge branch 'main' of https://github.com/openai/codex into shell-config

aibrahim-oai commented on codex-rs/core/src/context_manager/history_tests.rs:
```diff
@@ -11,9 +11,8 @@ use codex_protocol::models::ReasoningItemContent;
 use regex_lite::Regex;
-
-const EXEC_FORMAT_MAX_LINES: usize = 256;
 const EXEC_FORMAT_MAX_BYTES: usize = 10_000;
```
It's for the test

aibrahim-oai committed:
simplify

aibrahim-oai committed:
tests

aibrahim-oai committed:
tests

aibrahim-oai committed:
tests

aibrahim-oai commented:
@codex review this

aibrahim-oai committed:
lint

pakrym-oai commented on codex-rs/core/tests/suite/truncation.rs:
```diff
@@ -178,14 +177,8 @@ Output:
-397
-398
-399
-400
```
Can we keep the end?

pakrym-oai commented on codex-rs/core/tests/suite/shell_serialization.rs:
```diff
@@ -460,15 +459,10 @@ Output:
-
+\[…137224 tokens truncated…]
 .*
-396
```
let's bring this back

pakrym-oai commented on codex-rs/core/tests/suite/shell_serialization.rs:
```diff
@@ -460,15 +459,10 @@ Output:
 .*
-\[\.{3} omitted \d+ of 400 lines \.{3}\]
-
+\[…137224 tokens truncated…]
```
Should [ and ]be removed?

pakrym-oai commented on codex-rs/core/src/truncate.rs:
```diff
@@ -502,44 +358,24 @@ mod tests {
-        );
-        assert_ne!(truncated, large_error);
+        let truncated = truncate_text(&large_error, TruncationPolicy::Tokens(60));
+        dbg!(&truncated);
```
rem

pakrym-oai commented on codex-rs/core/src/truncate.rs:
```diff
@@ -550,34 +386,31 @@ mod tests {
-            truncate_with_line_bytes_budget(&content, model_format_max_bytes()),
-            content
-        );
+        assert_eq!(truncate_model_output(&content), content);
```
can we call the method being tested directly ?

pakrym-oai commented on codex-rs/core/tests/suite/user_shell_cmd.rs:
```diff
@@ -249,14 +249,11 @@ async fn user_shell_command_output_is_truncated_in_history() -> anyhow::Result<(
 
-    let head = (1..=128).map(|i| format!("{i}\n")).collect::<String>();
-    let tail = (273..=400).map(|i| format!("{i}\n")).collect::<String>();
-    let truncated_body =
```
Not truncated anymore?

pakrym-oai commented on codex-rs/core/src/context_manager/history_tests.rs:
we use tokens for tuncation then compare to bytes

aibrahim-oai committed:
tests

aibrahim-oai committed:
tests

aibrahim-oai committed:
out

aibrahim-oai committed:
out

aibrahim-oai committed:
lint

aibrahim-oai committed:
tests

pakrym-oai committed:
truncate: consolidate total-lines header parsing and adjust truncation tests

- Replace separate total-lines extraction and stripping with `strip_total_output_header` returning both body and count.
- Stop inserting newline between truncation marker and suffix, and update regexes/expectations to match new layout.
- Add and update truncation unit tests for bytes/tokens policies and UTF-8 handling.
- Tighten shell serialization and truncation suite tests, including smaller seq ranges

pakrym-oai committed:
Merge branch 'main' into shell-config

# Conflicts:
#	codex-rs/core/src/tools/events.rs
#	codex-rs/core/src/tools/mod.rs
#	codex-rs/core/src/truncate.rs

pakrym-oai committed:
tools: apply truncation policy to freeform shell output

Pass ToolEventCtx into format_exec_output_for_model and use
truncate_text with the turn truncation_policy for freeform shell
outputs. Update shell serialization tests to expect the new
truncation format and correct non-truncation boundary.

pakrym-oai committed:
tests: drop ShellCommand case from shell_output_reserializes_truncated_content

pakrym-oai committed:
Merge branch 'main' into shell-config

# Conflicts:
#	codex-rs/core/tests/suite/unified_exec.rs

pakrym-oai committed:
tests: fix truncation expectations for new tool output format

Adjust truncation tests to match new plain-text tool output:
- Update expected lengths and regex patterns for shell output truncation
- Fix typo in token count comment
- Set explicit tool_output_token_limit in MCP and user shell tests
- Tighten MCP truncation assertion with max length check

pakrym-oai committed:
tests: disable shell output truncation test on Windows
